jobs/clean_cubes.js

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable no-await-in-loop */
+// run with: node --max-old-space-size=8192 populate_analytics.js
+// will oom without the added tag
+
+// Load Environment Variables
+require('dotenv').config();
+
+const mongoose = require('mongoose');
+const Cube = require('../models/cube');
+const carddb = require('../serverjs/cards.js');
+
+const batchSize = 1000;
+
+const isInvalidCardId = (id) => carddb.cardFromId(id).name === 'Invalid Card';
+
+const needsCleaning = (cube) => {
+  if (!cube.cards) {
+    return true;
+  }
+
+  if (cube.cards.some((card) => !card || isInvalidCardId(card.cardID))) {
+    return true;
+  }
+
+  return false;
+};
+
+const processCube = async (leanCube) => {
+  if (needsCleaning(leanCube)) {
+    const cube = await Cube.findById(leanCube._id);
+
+    console.log(`Cleaning cube ${cube.name}: ${cube._id}`);
+
+    if (!cube.cards) {
+      cube.cards = [];
+    }
+    cube.cards = cube.cards.filter((c) => c && !isInvalidCardId(c.cardID));
+
+    await cube.save();
+  }
+};
+
+try {
+  (async () => {
+    await carddb.initializeCardDb();
+    mongoose.connect(process.env.MONGODB_URL).then(async () => {
```

Comment:
Since this is already an `async` function, we could save ourselves a level of nesting by just `await`ing the `connect` call and getting rid of the `then` block

Reply:
We should also pass `{ useNewUrlParser: true }` into the `connect` calls to get rid of mongoose warnings.